src/main/java/duke/tasks/TaskList.java

```diff
@@ -0,0 +1,248 @@
+        if (eventDescription.isBlank()) {
+            throw new DukeIllegalArgumentException(MISSING_ARGUMENT_MESSAGE);
+        }
+        String[] eventArgs = eventDescription.split(" /from ");
+        String[] eventTime = eventArgs[1].split(" /to ");
+        String startTime = eventTime[0];
+        String endTime = eventTime[1];
+        Task event = new Event(eventArgs[0], startTime, endTime);
+        taskList.add(event);
+        updateStorage();
+        return String.format(ADD_TASK_MESSAGE, event, taskList.size());
+    }
+
+    /**
+     * Marks a task in TaskList as completed.
+     * @param taskIndex The index of the task in TaskList.
+     * @return String with return message.
+     * @throws DukeIllegalArgumentException If the taskIndex is missing.
+     */
+    public String markTask(String taskIndex) {
+        if (taskIndex.isBlank()) {
+            throw new DukeIllegalArgumentException(MISSING_ARGUMENT_MESSAGE);
+        }
+        Task t = taskList.get(Integer.parseInt(taskIndex) - 1);
+        t.markAsDone();
+        updateStorage();
+        return String.format(MARK_DONE_MESSAGE, t);
+    }
+
+    /**
+     * Unmarks a task in TaskList as not completed.
+     * @param taskIndex The index of the task in TaskList.
+     * @return String with return message.
+     * @throws DukeIllegalArgumentException If the taskIndex is missing.
+     */
+    public String unmarkTask(String taskIndex) {
+        if (taskIndex.isBlank()) {
+            throw new DukeIllegalArgumentException(MISSING_ARGUMENT_MESSAGE);
+        }
+        Task t = taskList.get(Integer.parseInt(taskIndex) - 1);
+        t.markAsUndone();
+        updateStorage();
+        return String.format(MARK_UNDONE_MESSAGE, t);
+    }
+
+    /**
+     * Deletes a task in TaskList.
+     * @param taskIndex The index of the task to be deleted in TaskList.
+     * @return String with return message.
+     * @throws DukeIllegalArgumentException If the taskIndex is missing.
+     */
+    public String deleteTask(String taskIndex) {
+        if (taskIndex.isBlank()) {
+            throw new DukeIllegalArgumentException(MISSING_ARGUMENT_MESSAGE);
+        }
+        Task taskToRemove = taskList.get(Integer.parseInt(taskIndex) - 1);
+        taskList.remove(Integer.parseInt(taskIndex) - 1);
+        updateStorage();
+        return String.format(REMOVE_TASK_MESSAGE, taskToRemove, taskList.size());
+    }
+
+    /**
+     * Finds tasks that contains the keyword given.
+     * @param keyword Keyword to find in the taskList.
+     * @return String with return message.
+     * @throws DukeIllegalArgumentException If the keyword is missing.
+     */
+    public String findTask(String keyword) {
+        if (keyword.isBlank()) {
+            throw new DukeIllegalArgumentException(MISSING_ARGUMENT_MESSAGE);
+        }
+        StringBuilder sb = new StringBuilder();
+        int count = 1;
+        for (Task task : taskList) {
+            if (task.description.contains(keyword)) {
```

Comment:
Perhaps there could be a method within `Task`, `doesDescriptionContains(String keyword)`, that checks if the task's description contains the keyword, instead of exposing the `description` outside of the abstraction barrier?